app/controllers/admin/notices_controller.rb

```diff
@@ -0,0 +1,57 @@
+class Admin::NoticesController < ApplicationController
+  load_permissions_and_authorize_resource
+  before_action :authorize
+
+  def index
+    @notice_grid = initialize_grid(Notice)
+  end
+
+  def new
+    @notice = Notice.new
+  end
+
+  def edit
+    @notice = Notice.find(params[:id])
+  end
+
+  def create
+    @notice = Notice.new(notice_params)
+    @notice.user = current_user
+    if @notice.save
+      redirect_to edit_admin_notice_path(@notice),
+                  notice: alert_create(Notice)
+    else
+      render :new, status: 422
+    end
+  end
+
+  def update
+    @notice = Notice.find(params[:id])
+    @notice.user = current_user
+    if @notice.update(notice_params)
+      redirect_to edit_admin_notice_path(@notice),
+                  notice: alert_update(Notice)
+    else
+      render :edit, status: 422
+    end
+  end
+
+  def destroy
+    notice = Notice.find(params[:id])
+
+    notice.destroy!
+    redirect_to admin_notices_url, notice: alert_destroy(Notice)
+  end
+
+  private
+
+  def authorize
+    can?(:modify, Notice)
```

Comment:
Is this correct? So you need modify permission also to read the notices? And is not your load_permissions_and_authorize_resource taking care of this at the top?